### DISCUSSION TITLE
Boot problems with i20230828

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1411] [Fixed Issues]
Hi

Has anybody tried installing the new i20230828 image on RPi 3B ?  I downloaded this version this morning and flashed it to a new SD card.  On first boot it seemed to work OK but I needed to turn off Bluetooth (which is enabled by default in this version) which required a reboot.  Following the reboot the system was completely unresponsive - via web app or ssh.  As I can't access the system to check for logs etc I cannot give any more information at the moment.

Chris

**Comment by hjfisch** [comment]
I've 2 Rpi 3b .
New install on both without any probs .
One on sd-card and one on usb-stick .
Maybe you should try another sd-card .

**Comment by ceejayemm** [comment]
Not sure the card was at fault as the initial boot would succeed but fail on a reboot, but I have flashed another with i20230828 and this time it seems to start up ok.  

There is however another problem with this build in that it is not possible to enable the Browser on RPi feature. It appears  there are errors starting Chromium (which is an old error seen many time before in the past):

Sep 01 16:30:35 RAConservatory sudo[2404]:     root : PWD=/ ; USER=root ; COMMAND=/usr/bin/startx
Sep 01 16:30:35 RAConservatory sudo[2404]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Sep 01 16:30:36 RAConservatory sudo[2423]: X.Org X Server 1.21.1.8
Sep 01 16:30:36 RAConservatory sudo[2423]: X Protocol Version 11, Revision 0
Sep 01 16:30:36 RAConservatory sudo[2423]: Current Operating System: Linux RAConservatory 6.1.47-1-rpi-ARCH #1 SMP PREEMPT Fri Aug >
Sep 01 16:30:36 RAConservatory sudo[2423]: Kernel command line: coherent_pool=1M 8250.nr_uarts=0 snd_bcm2835.enable_headphones=0 bc>
Sep 01 16:30:36 RAConservatory sudo[2423]:
Sep 01 16:30:36 RAConservatory sudo[2423]: Current version of pixman: 0.42.2
Sep 01 16:30:36 RAConservatory sudo[2423]:         Before reporting problems, check http://wiki.x.org
Sep 01 16:30:36 RAConservatory sudo[2423]:         to make sure that you have the latest version.
Sep 01 16:30:36 RAConservatory sudo[2423]: Markers: (--) probed, (**) from config file, (==) default setting,
Sep 01 16:30:36 RAConservatory sudo[2423]:         (++) from command line, (!!) notice, (II) informational,
Sep 01 16:30:36 RAConservatory sudo[2423]:         (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
Sep 01 16:30:36 RAConservatory sudo[2423]: (==) Log file: "/var/log/Xorg.0.log", Time: Fri Sep  1 16:30:36 2023
Sep 01 16:30:36 RAConservatory sudo[2423]: (==) Using config directory: "/etc/X11/xorg.conf.d"
Sep 01 16:30:36 RAConservatory sudo[2423]: (==) Using system config directory "/usr/share/X11/xorg.conf.d"
Sep 01 16:30:36 RAConservatory sudo[2424]: modprobe: FATAL: Module g2d_23 not found in directory /lib/modules/6.1.47-1-rpi-ARCH
Sep 01 16:30:36 RAConservatory sudo[2429]: Failed to connect to bus: No medium found
Sep 01 16:30:36 RAConservatory sudo[2430]: dbus-update-activation-environment: error: unable to connect to D-Bus:
Using X11 for dbus-daemon autolaunch was disabled at compile time, set your DBUS_SESSION_BUS_ADDRESS instead
Sep 01 16:30:36 RAConservatory sudo[2443]: /usr/lib/chromium/chromium: error while loading shared libraries:
libre2.so.10: cannot open shared object file: No such file or directory
Sep 01 16:30:36 RAConservatory sudo[2422]: xinit: connection to X server lost
Sep 01 16:30:36 RAConservatory sudo[2422]:
Sep 01 16:30:36 RAConservatory sudo[2422]: waiting for X server to shut down
Sep 01 16:30:36 RAConservatory sudo[2423]: (II) Server terminated successfully (0). Closing log file.
Sep 01 16:30:36 RAConservatory sudo[2404]: pam_unix(sudo:session): session closed for user root
Sep 01 16:30:36 RAConservatory systemd[1]: localbrowser.service: Deactivated successfully.
-- Subject: Unit succeeded


Also there is a Python error later in the start sequence:

Sep 01 16:56:39 RAConservatory mpdidle.sh[3055]: /srv/http/bash/common.sh: line 151: echo: write error: Broken pipe
Sep 01 16:56:39 RAConservatory mpdidle.sh[3055]: /srv/http/bash/status.sh: line 20: echo: write error: Broken pipe
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]: connection handler failed
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]: Traceback (most recent call last):
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py",
line 1314, in close_connection
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     await self.transfer_data_task
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py",
line 968, in transfer_data
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     message = await self.read_message()
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:               ^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py",
line 1038, in read_message
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     frame = await self.read_data_frame(max_size=self.max_size)
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py",
line 1113, in read_data_frame
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     frame = await self.read_frame(max_size)
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py",
line 1170, in read_frame
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     frame = await Frame.read(
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:             ^^^^^^^^^^^^^^^^^
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/framing.py",
line 69, in read
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     data = await reader(2)
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:            ^^^^^^^^^^^^^^^
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/asyncio/streams.py", line 729, in readexactly
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     await self._wait_for_data('readexactly')
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/asyncio/streams.py", line 522, in _wait_for_data
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     await self._waiter
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]: asyncio.exceptions.CancelledError
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]: The above exception was the direct cause of the following exception:
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]: Traceback (most recent call last):
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/server.py", 
line 236, in handler
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     await self.ws_handler(self)
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/srv/http/bash/cmd-websocket.py", line 8, in cmd
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     async for args in websocket:
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py", 
line 497, in __aiter__
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     yield await self.recv()
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:           ^^^^^^^^^^^^^^^^^
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py", 
line 568, in recv
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     await self.ensure_open()
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:   File "/usr/lib/python3.11/site-packages/websockets/legacy/protocol.py", 
line 953, in ensure_open
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]:     raise self.connection_closed_exc()
Sep 01 16:58:04 RAConservatory cmd-websocket.py[1151]: websockets.exceptions.ConnectionClosedError: sent 1011 (unexpected error) 
keepalive ping timeout; no close frame received
Sep 01 16:59:19 RAConservatory sudo[3206]:     http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/status.sh withdisplay
Sep 01 16:59:19 RAConservatory sudo[3206]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Sep 01 16:59:19 RAConservatory sudo[3215]:     http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/status.sh withdisplay
Sep 01 16:59:19 RAConservatory sudo[3215]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Sep 01 16:59:20 RAConservatory sudo[3206]: pam_unix(sudo:session): session closed for user root
Sep 01 16:59:20 RAConservatory sudo[3215]: pam_unix(sudo:session): session closed for user root
Sep 01 16:59:23 RAConservatory cmd-websocket.py[1151]: connection handler failed

These errors are repeated several times after this.

Chris

**Reply by rern** [reply]
Hi Chris,

Chromium on 64bit version is now gone with the one on 32bit which has been broken since April.
Seem like Firefox should rule them all. Please `pacman -Sy firefox` and it will be ok again.

**Reply by ceejayemm** [reply]
@rern 

Our updates crossed.  

Thanks, the change to Firefox seems to have solved the RPi on Browser' issue.  It did also seem to update a whole load of 'keys' along the way. I have many instances (all different) of:

Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[5611]: Refreshing key F850562FCDA369F80D33000AE48D0A8326DE47C5 with UID morganamilo@archlinux.org>
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]: pub   rsa4096 2021-07-10 [SC]
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]:       F850562FCDA369F80D33000AE48D0A8326DE47C5
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]: uid           [  full  ] Morgan Adamiec <morganamilo@archlinux.org>
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]: uid           [ unknown] Morgan Adamiec <morgan@grdigital.co.uk>
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]: uid           [ unknown] Morgan Adamiec <morganamilo@gmail.com>
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]: sub   rsa4096 2021-07-29 [A]
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[6476]: sub   rsa4096 2021-07-10 [E]
Sep 02 10:13:52 RAConservatory archlinux-keyring-wkd-sync[5611]: Skipping key F850562FCDA369F80D33000AE48D0A8326DE47C5 with UID morgan@grdigital.co.uk...

Are the Python errors also related to this ? Also there are a number of PHP errors like:

*177 FastCGI sent in stderr: "; PHP message: PHP Warning:  Undefined array key "returnhtml"

See error lines at Sep 02 10:17:04 in the attached file which contains out put from journalctl -b

[journal3.txt.zip](https://github.com/rern/rAudio/files/12503675/journal3.txt.zip)

Chris

**Comment by kw-sudo** [comment]
**Image 20230828@RPi 3B+**

With the current image, I felt like I 'had' to set up a new installation. No pacman -Syu, only r/Audio.

And, I hardly dare to say it, everything went flawlessly.

@ceaajymem The only customization of the image was the _config.txt_ before first run (...and had to deactivate WiFi afterwards in settings):

`disable_overscan=1
disable_splash=1

gpu_mem=16
max_usb_current=1
dtparam=krnbt=off
dtparam=audio=off
hdmi_blanking=1
hdmi_ignore_hotplug=1
hdmi_ignore_composite=1

dtparam=act_led_trigger=none
dtparam=act_led_activelow=off

dtparam=pwr_led_trigger=none
dtparam=pwr_led_activelow=off

dtoverlay=disable-wifi
dtoverlay=disable-bt`

**Comment by ceejayemm** [comment]
@master-kw 

The install and basic working of this image (eventually) went fine fine and basic functionality is OK.  I have one minor change to config.txt which is:

dtparam=i2c_vc_baudrate=50000

which is the Raspberry.org workaround for using an official RPi touchscreen with RPi3B (see https://github.com/raspberrypi/linux/issues/4686), and I have been using it for over a year with with rAudio with no problems.

It was only when I tried enabling 'Browser on RPi' that things started going wrong.  Using 'journalctl -xe' I found the errors above relating to the start up of Chromium. I then started seeing the large numbers of Python errors- which I don't know what these relate to.

Chris

**Comment by ceejayemm** [comment]
As per #1647 the remaining websocket-server issue seems to be resolved with 20240315.

Chris
